web/src/main/java/org/springframework/security/web/server/authentication/session/PreventLoginMaximumSessionsExceededHandler.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.authentication.session;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.web.authentication.session.SessionAuthenticationException;
+
+/**
+ * Throws a {@link SessionAuthenticationException} when the maximum number of sessions for
+ * a user has been reached.
+ *
+ * @author Marcus da Coregio
+ * @since 6.3
+ */
+public final class PreventLoginMaximumSessionsExceededHandler implements ServerMaximumSessionsExceededHandler {
```

Comment:
I'd recommend the word `Server` also be in the class name. The reason is if we were to add the same thing on the servlet side, we want the class names to be different for clarity.